### PR TITLE
Top fixes

### PR DESCRIPTION
**Pull request by abudnik**
Fixes for pull request (top keys statistsics): https://github.com/reverbrain/elliptics/pull/573

### CODE REVIEW

File: monitor/monitor.h [hunk 1]
```diff
@@ -106,9 +106,9 @@ void dnet_monitor_remove_provider(struct dnet_node *n, const char *name);
  * \a size - size of data that takes a part in command execution
  * \a time - time spended on command execution
  */
-void monitor_command_counter(struct dnet_node *n, const int cmd, const int trans,
-                             const int err, const int cache,
-                             const uint32_t size, const unsigned long time);
+void dnet_monitor_stats_update(struct dnet_node *n, const struct dnet_cmd *cmd, const int trans,
```
Comment (bioothod) [inline monitor/monitor.h]: Why do we need transaction number (as int, is this is an error?) when we have the whole command structure?

File: monitor/top_stats.cpp [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+void top_stats::update_stats(const struct dnet_cmd *cmd, uint64_t size)
+{
+	if (size > 0 && cmd->cmd == DNET_CMD_READ) {
+		key_stat_event event(cmd->id, size, 1., time(nullptr));
+		m_stats.add_event(event, event.get_time());
+	}
+}
+
+}} /* namespace ioremap::monitor */
+
```
Comment (bioothod) [inline monitor/top_stats.cpp]: Please merge top stats and provider interface into common 'top' file, do not split it into single-three-lines-function chunks

File: tests/pytests/test_monitor_top.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+def check_keys_fields(keys):
+    '''all items in top key list must contain group_id, key_id, traffic size generated by key access, key access frequency'''
+    for k in keys:
+        assert k['group']
+        assert k['id']
+        assert k['size']
+        assert k['frequency']
+
+'''
+Test fact of top keys service existance, service work and protocol correctness
```
Comment (bioothod) [inline tests/pytests/test_monitor_top.py]: How this test does this? Please describe in more details what test is supposed to run, what results should be obtained and how they are going to be checked

File: tests/stats_test.cpp [hunk 4]
```diff
@@ -0,0 +1,415 @@
...
+	auto top_stats = monitor->get_top_stats();
+	BOOST_CHECK(top_stats != nullptr);
+}
+
+/****************
+ Test event_stats
+ ****************/
+typedef ioremap::monitor::event_stats<test_event> stats_t;
+
+static void test_event_stats_boundary_conditions()
```
Comment (bioothod) [inline tests/stats_test.cpp]: Please describe these tests with more details, similar to what I have asked for python test

File: tests/stats_test.cpp [hunk 5]
```diff
@@ -0,0 +1,415 @@
...
+					        snd.begin(), snd.end(),
+					        diff.begin(), test_event::key_compare_event);
+	return std::distance(diff.begin(), it);
+}
+
+/*
+ * Check max-heap property of event_stats. All events inserted at the same time.
+ * All events describe access to different keys.
+ */
+static void test_event_stats_no_time_dependency()
```
Comment (bioothod) [inline tests/stats_test.cpp]: This needs more details - what this test is supposed to check, how you do this, what results are going to be received, how will you interpret them, how will you check that they are correct or wrong and so on

File: tests/pytests/test_monitor_top.py (hunk 3 above)
Comment (abudnik) [inline tests/pytests/test_monitor_top.py]: https://github.com/abudnik/elliptics/commit/ad8bfc5e4e8f674b3e1c2638fab3cc937a4168a5

File: tests/stats_test.cpp (hunk 5 above)
Comment (abudnik) [inline tests/stats_test.cpp]: https://github.com/abudnik/elliptics/commit/470ba9df7ab1060717b72bc21c2b16d31f1f948b

**Comment by abudnik** [comment]
Is there anything to fix or refine?

**Comment by bioothod** [comment]
Looks good to me, is it ready for merge?

**Comment by abudnik** [comment]
Yes, it is ready to merge.

**Comment by bioothod** [comment]
I will make a new version release

**Pull request merged**